### PR TITLE
Add documentation for the new Weekly InstructLab CLI Meeting

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Text is based off the existing text for the `Weekly All Community Meeting` but definitely open to suggestions!

### CODE REVIEW

File: Collaboration.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ Weekly community meetings will feature regular updates on project happenings, in
 
 If you are new to the InstructLab project, the weekly community meeting is a great place to get started!
 
+### [Weekly InstructLab CLI Meeting](#weekly-instructlab-cli-meeting)
+
+We host weekly InstructLab meetings each Thursday at 14:00 UTC. ([time zone converter](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2024&month=5&day=14&hour=14&min=0&sec=0&p1=37&p2=43&p3=101&p4=224&p5=213&p6=771&p7=248&p8=2))
+
+Weekly InstructLab CLI meetings will feature announcements from the [CLI Maintainers](MAINTAINERS.md) team about the CLI project, overviews of the general roadmap the project is taking via our [Milestones](https://github.com/instructlab/instructlab/milestones) as well as the [Project Board](https://github.com/orgs/instructlab/projects/2), and dedicated time for members of community comment on and contribute to the direction of the project.
```
Comment (russellb) [inline Collaboration.md:27]: nit ... someday i'd like to turn on some kind of reasonable line length limit in markdown linting. When paragraphs are all on a single line, it makes diffs harder to read. Using an editor that can do the line wrapping automatically makes this not a big deal to enforce.

Comment (russellb) [inline Collaboration.md:27]:
```suggestion
Weekly InstructLab CLI meetings will feature announcements from the [CLI Maintainers](MAINTAINERS.md) team about the CLI project, overviews of the general roadmap the project is taking via our [Milestones](https://github.com/instructlab/instructlab/milestones) as well as the [Project Board](https://github.com/orgs/instructlab/projects/2), and dedicated time for discussing the direction of the project.
```

**Review by russellb — CHANGES_REQUESTED** [review]
minor wording suggestion where it seemed like a couple words were missing

File: Collaboration.md (hunk 1 above)
Comment (nathan-weinberg) [reply Collaboration.md]: Would not be opposed to that - especially in docs-heavy repos like `community` - but probably won't happen in this PR

**Review by lhawthorn — APPROVED** [review]
Noting @russellb's feedback on maximum line length linting for Markdown files - and this should be opened as a separate issue  (if you would please do the honors, @nathan-weinberg) - this looks good to me.

Please rebase, @nathan-weinberg, so I can take one last look. 

And noting to myself this should have been a review requesting changes and not a review with approval, which is my mistake. I am not certain how to fix that retroactively, but will investigate for future learnings.

File: Collaboration.md (hunk 1 above)
Comment (nathan-weinberg) [reply Collaboration.md]: @lhawthorn @russellb opened https://github.com/instructlab/community/issues/268 to follow up on this

**Comment by nathan-weinberg** [comment]
@Mergifyio rebase

File: Collaboration.md [hunk 2]
```diff
@@ -20,6 +20,14 @@ Weekly community meetings will feature regular updates on project happenings, in
 
 If you are new to the InstructLab project, the weekly community meeting is a great place to get started!
 
+### [Weekly InstructLab CLI Meeting](#weekly-instructlab-cli-meeting)
+
+We host weekly InstructLab meetings each Thursday at 14:00 UTC. ([time zone converter](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2024&month=5&day=14&hour=14&min=0&sec=0&p1=37&p2=43&p3=101&p4=224&p5=213&p6=771&p7=248&p8=2))
```
Comment (lhawthorn) [inline Collaboration.md:25]: Propose rewording to "We host weekly InstructLab CLI meetings" as we are now running into the tool has the same name as the project and can cause confusion problem.

**Review by lhawthorn — CHANGES_REQUESTED** [review]
We seem to have lost @russellb's original wording change suggestions, so I would like him to take another look before we merge.

**Review by russellb — APPROVED** [review]
lgtm

**Review by lhawthorn — APPROVED** [review]
LGTM too - merging.

**Pull request merged**